File: test_mcp_sse.py

import urllib.request
import json
import time

def test_mcp_call():
    print("--- Thử gọi add_episode trực tiếp (Python v2) ---")
    url = "http://localhost:8000/sse"
    
    try:
        req = urllib.request.Request(url)
        with urllib.request.urlopen(req, timeout=10) as response:
            message_url = None
            for line_bytes in response:
                if line_bytes:
                    line_str = line_bytes.decode('utf-8').strip()
                    if line_str.startswith("data: /messages/"):
                        message_url = f"http://localhost:8000{line_str.replace('data: ', '').strip()}"
                        print(f"Tìm thấy endpoint: {message_url}")
                        break
            
            if not message_url: return

            # Thử gọi tool add_episode
            print("Đang gửi yêu cầu tools/call (add_episode)...")
            payload = json.dumps({
                "jsonrpc": "2.0",
                "id": 2,
                "method": "tools/call",
                "params": {
                    "name": "add_episode",
                    "arguments": {
                        "content": "Đây là một bài kiểm tra ghi nhớ từ Antigravity lúc " + str(time.time())
                    }
                }
            }).encode('utf-8')

            post_req = urllib.request.Request(message_url, data=payload, headers={'Content-Type': 'application/json'})
            urllib.request.urlopen(post_req, timeout=10)
            
            print("Đang chờ phản hồi...")
            for line_bytes in response:
                if line_bytes:
                    line_str = line_bytes.decode('utf-8').strip()
                    if line_str.startswith("data:"):
                        data_str = line_str.replace("data: ", "").strip()
                        print(f"Nhận được: {data_str}")
                        if '"id":2' in data_str:
                            if "error" in data_str:
                                print("Lỗi từ server khi gọi tool.")
                            else:
                                print("--- THÀNH CÔNG! Tool add_episode đã được gọi. ---")
                            return
    except Exception as e:
        print(f"Lỗi: {e}")

if __name__ == "__main__":
    test_mcp_call()

File: test_mcp_sse.js

const EventSource = require('eventsource');
const axios = require('axios');

async function testMCP() {
    console.log('--- Đang kết nối tới Graphiti MCP SSE ---');
    const es = new EventSource('http://localhost:8000/sse');
    
    let messageEndpoint = '';
    
    es.addEventListener('endpoint', async (event) => {
        messageEndpoint = 'http://localhost:8000' + event.data;
        console.log('Nhận được endpoint tin nhắn:', messageEndpoint);
        
        try {
            console.log('Đang gửi yêu cầu listing tools...');
            const response = await axios.post(messageEndpoint, {
                jsonrpc: '2.0',
                id: 1,
                method: 'tools/list'
            });
            console.log('Yêu cầu POST thành công (HTTP 202/Accepted)');
        } catch (error) {
            console.error('Lỗi khi gửi POST:', error.message);
        }
    });

    es.onmessage = (event) => {
        try {
            const data = JSON.parse(event.data);
            if (data.result && data.result.tools) {
                console.log('--- Danh sách công cụ tìm thấy: ---');
                data.result.tools.forEach(tool => {
                    console.log(` - ${tool.name}: ${tool.description}`);
                });
                es.close();
                process.exit(0);
            } else {
                console.log('Nhận được dữ liệu SSE:', event.data);
            }
        } catch (e) {
            console.log('Nhận được dữ liệu thô:', event.data);
        }
    };

    es.onerror = (err) => {
        console.error('Lỗi SSE:', err);
        es.close();
        process.exit(1);
    };

    // Timeout sau 15 giây
    setTimeout(() => {
        console.log('Hết thời gian chờ phản hồi tools/list');
        es.close();
        process.exit(1);
    }, 15000);
}

testMCP();
